providers/flagd/pkg/provider.go

```diff
@@ -72,6 +73,13 @@ func NewProvider(opts ...ProviderOption) (*Provider, error) {
 			CustomSyncProviderUri:   provider.providerConfiguration.CustomSyncProviderUri,
 			GrpcDialOptionsOverride: provider.providerConfiguration.GrpcDialOptionsOverride,
 		})
+		provider.hooks = append(provider.hooks, NewSyncContextHook(
```

Comment:
I am pretty sure, that this could be less complex with go native approach. please teach me